etcd/cluster.go

```diff
@@ -3,12 +3,14 @@ package etcd
 import (
 	"math/rand"
 	"strings"
+	"sync"
 )
 
 type Cluster struct {
 	Leader   string   `json:"leader"`
 	Machines []string `json:"machines"`
 	picked   int
+	pickLock *sync.RWMutex
```

Comment:
you need to use `sync.RWMutex` instead of `*sync.RWMutex`.
could you rename to `mu` so it is a general one that protects both machines and picked?